src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+
+public class Contact {
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) {
```

Comment:
Good attempt at extracting the Contact class into a superclass, However I noticed that you have left out the validation check for each of the subclasses. As discussed in the tutorial, it is also possible to have the validation methods within the superclass such that each of the subclasses have their inputs properly validated